lib/src/auth/exception/auth_exception.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:firedart/src/auth/mapper/auth_error_code_mapper.dart';
+
+/// Exception thrown when the authentication is failed.
+class AuthException implements Exception {
+  /// The error code of the exception.
```

Comment:
```suggestion
  /// An error code of this exception.
```
Consider this lint: https://dart.dev/guides/language/effective-dart/documentation#prefer-using-this-instead-of-the-to-refer-to-a-members-instance